test/test_python_dispatch.py

```diff
@@ -641,6 +641,52 @@ def test_autograd_in_attr(self):
         self.assertIsNone(t.grad)
         self.assertIsNotNone(t.elem.grad)
 
+    def test_dispatch_super_call(self):
+        called = []
+
+        class SubTensor(torch.Tensor):
+            @staticmethod
+            def __new__(cls, elem):
+                return torch.Tensor._make_subclass(cls, elem)
+
+            __torch_function__ = torch._C._disabled_torch_function_impl
```

Comment:
question unrelated to this PR - right now if you want to "just use torch_dispatch", you need to manually disable torch_function like the above.

It seems like using both of them is the exception rather than the common case. Do you think we'd wanna consider flipping the default so that torch_function is default disabled for your subclass if you write your own torch_dispatch? (Open question around how BC-breaking that is)

Reply:
We should do this, and I think literally no one will be BC broken by it.